llvm/lib/Transforms/Instrumentation/GCOVProfiling.cpp

```diff
@@ -803,6 +803,8 @@ bool GCOVProfiler::emitProfileNotes(
         continue;
       if (F.hasFnAttribute(llvm::Attribute::SkipProfile))
         continue;
+      if (F.hasFnAttribute(llvm::Attribute::DisableSanitizerInstrumentation))
```

Comment:
This one is not about sanitizers.

Reply:
I see why we need this, it's just a little bit inconsistent to use attribute with sanitizer in name to suppress profiling.
would be possible to add a test for GCOV and memprof as well?

Reply:
As Vitaly said this is not a sanitizer, so it'll be unintuitive if the attribute also kills instrumentation here.

For code that needs to disable all kinds of instrumentation, something like Linux's `noinstr` is needed, and you have to build a list of attributes that also includes `__attribute__((no_profile_instrument_function))`.

Reply:
Thanks for the feedback, I move all profiling to a separate PR.
`no_profile_instrument_function` sound better for profiles as a common attribute.